### PR TITLE
fix: per diem page to use the advanced_per_diem_settings

### PR DESCRIPTION
**Pull request by arjunaj5**
## Clickup
https://app.clickup.com

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

## Summary by CodeRabbit

- **New Features**
	- Introduced advanced per diem settings, allowing for more granular control over employee restrictions.
	- Enhanced organizational settings for managing per diem allowances.
	- Added error handling for the `num_days` input field to improve user feedback.

- **Bug Fixes**
	- Improved error handling for policy violations during expense creation and editing.

- **Tests**
	- Expanded test cases for the `AddEditPerDiemPage` component to validate new per diem settings. 

- **Documentation**
	- Updated internal documentation to reflect changes in organizational settings related to per diem management.

### CODE REVIEW

File: src/app/core/services/org-settings.service.ts [hunk 1]
```diff
@@ -157,6 +157,11 @@ export class OrgSettingsService {
         enable_category_restriction:
           incoming.advanced_project_settings && incoming.advanced_project_settings.enable_category_restriction,
       },
+      advanced_per_diems_settings: {
+        allowed: incoming.advanced_per_diems_settings?.allowed,
+        enabled: incoming.advanced_per_diems_settings?.enabled,
+        enable_employee_restriction: incoming.advanced_per_diems_settings?.enable_employee_restriction,
+      },
```
[the remark this answers is not in the document]
Comment (Chethan-Fyle) [reply src/app/core/services/org-settings.service.ts:164]: +1

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts [hunk 2]
```diff
@@ -959,7 +959,7 @@
       sub_category: [],
       per_diem_rate: [, Validators.required],
       purpose: [],
-      num_days: [, Validators.compose([Validators.required, Validators.min(0)])],
+      num_days: [, Validators.compose([Validators.required, Validators.min(0), Validators.pattern(/^\d+$/)])],
```
[the remark this answers is not in the document]
Comment (Chethan-Fyle) [reply src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts:962]: +1

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts [hunk 3]
```diff
@@ -1224,7 +1224,9 @@
 
             if (expenseField.is_mandatory) {
               if (txnFieldKey === 'num_days') {
-                control.setValidators(Validators.compose([Validators.required, Validators.min(0)]));
+                control.setValidators(
+                  Validators.compose([Validators.required, Validators.min(0), Validators.pattern(/^\d+$/)])
+                );
```
[the remark this answers is not in the document]
Comment (Chethan-Fyle) [reply src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts:1229]: +1

**Review by Chethan-Fyle — CHANGES_REQUESTED** [review]
Few comments.

**Review by Chethan-Fyle — APPROVED** [review]

**Review by mvaishnavi — APPROVED** [review]

**Pull request merged**